ibm/mas_devops/roles/mongodb/tasks/providers/community/check-mongo-exists.yml

```diff
@@ -38,10 +38,30 @@
       set_fact:
         existing_mongo_minor_version: "{{ existing_mongo_version | regex_search('(?<=)(.*)(?=...)') }}"
 
+    - include_tasks: tasks/determine-ibmcatalog-tag.yml
+
+    - name: Check if file exists in casebundles
+      stat:
+        path: "{{ role_path }}/../../common_vars/casebundles/{{ catalog_tag }}.yml"
+      register: stat_result
+      when: catalog_tag is defined and catalog_tag != ""
+
+    - name: Load CASE bundle versions
+      include_vars:
+        file: "{{ role_path }}/../../common_vars/casebundles/{{ catalog_tag }}.yml"
+      when: stat_result is defined and stat_result.stat.exists
+
     # holds the expected target mongo version that is feature compatible with the existing mongo instance version
+    # mongo extras version are picked up from the catalog if available or uses defaults.
     - name: Set mongo_compatible_target_version
       set_fact:
         mongo_compatible_target_version: "{{ mongo_feature_compatibility_matrix[existing_mongo_minor_version] }}"
+      vars:
+        mongo_feature_compatibility_matrix:
```

Comment:
I think having this var in the `defaults/main.yml` would be easier to find out an maintain going forward, no? just a thought...

Also, do we need to set and maintain a default value here for `mongo_extras_version_X` ? We already define the default for each version in the corresponding casebundle file, so we'll always have a value coming from there, no?

Reply:
@andrercm am using the mongo_extras_version_X value from the case bundle vars..

Reply:
I ll move the defaults back to defaults/main.yml